frontend/src/metabase-types/api/mocks/settings.ts

```diff
@@ -131,7 +131,7 @@ export const createMockTokenFeatures = (
 export const createMockSettingDefinition = (
   opts?: Partial<SettingDefinition>,
 ): SettingDefinition => ({
-  key: "key",
+  key: "active-users-count",
```

Comment:
Doesn't this risk making some test mis-behave? I think it was intentional that it was a non existing key so that we didn't override some real setting my mistake.
I think it's fine if we use "as XXX" in mock functions

Reply:
I think technically, `key` shouldn't be optional. We should fix this by making `key` mandatory. I doubt there would be a call to `createMockSettingDefinition` without specifying a `key` though.

Reply:
I've done that in this PR and fixed up the issues that arose from that. I think it should be good now